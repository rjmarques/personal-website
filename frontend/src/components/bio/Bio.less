@import "../../theme.less";

.bio {
  margin-top: 40px;

  .bio-grid {
    display: grid;
    grid-template-columns: 1fr;
    position: relative;

    & > :first-child {
      display: none;
    }

    .for-size(laptop-up; {
      & > :first-child {
        display: block
      }

      grid-template-columns: 1fr 5fr;
    });
  }

  .bio-text {
    .for-size(laptop-up; {padding-left: 80px;});
  }

  .bio-picture {
    img {
      background-color: white;
      border-radius: 100%;
      border: 4px solid @light-grey;
      height: 150px;
      object-fit: cover;
      object-position: 8px 10px;
      width: 150px;
    }
  }

  .bio-divider {
    background: @light-grey;
    display: none;
    height: 100%;
    position: absolute;
    width: 2px;

    .for-size(laptop-up; {display: block; margin-left: 35px;});
    .for-size(desktop-up; {display: block; margin-left: 20px;});
  }
}
