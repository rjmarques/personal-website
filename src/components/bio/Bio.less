@import "../../theme.less";

.Bio {
  margin-top: 40px;
}

.Bio-text {
  border-left: 1px solid @light-grey;
  padding-left: 45px;
}

.Bio-picture {
  img {
    border-radius: 100%;
    border: 4px solid @light-grey;
    height: 150px;
    width: 150px;
  }
}
