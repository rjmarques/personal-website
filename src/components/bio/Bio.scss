@use "../../theme.scss" as *;

.bio {
  margin-top: 40px;

  .bio-grid {
    display: grid;
    grid-template-columns: 1fr;
    position: relative;

    & > :first-child {
      display: none;
    }

    @include for-size(laptop-up) {
      & > :first-child {
        display: block;
      }

      grid-template-columns: 1fr 5fr;
    }
  }

  .bio-text {
    @include for-size(laptop-up) {
      padding-left: 80px;
    }
  }

  .bio-picture {
    img {
      background-color: white;
      border-radius: 100%;
      border: 4px solid $light-grey;
      height: 150px;
      object-fit: cover;
      object-position: 8px 10px;
      width: 150px;
    }
  }

  .bio-divider {
    background: $light-grey;
    display: none;
    height: 100%;
    position: absolute;
    width: 2px;

    @include for-size(laptop-up) {
      display: block;
      margin-left: 35px;
    }
    @include for-size(desktop-up) {
      display: block;
      margin-left: 20px;
    }
  }
}
