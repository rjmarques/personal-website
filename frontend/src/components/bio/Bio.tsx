import React from "react";

import ContentSection from "../common/ContentSection";

import "./Bio.less";

const Bio = () => (
  <ContentSection
    title="about me"
    subTitle="Know me a bit better"
    wrapperClass="bio"
  >
    <div className="bio-grid">
      <div className="bio-picture">
        <img alt="ric" src={require("../../assets/img/me.jpeg")} />
      </div>
      <div>
        <div className="bio-divider" />
        <div className="bio-text materialize">
          <p>
            Hi, I'm Ricardo Marques. I'm a professional nerd that works on both
            the back and front end, while using a modern tech stack to make
            things go: <i>Beep-Boop</i>.
          </p>
          <p>
            I started off, as a Java developer, building JEE applications. From
            multithread solutions, to REST integrations and even JPA enabled
            EJBs. However, I've always had a big interest in all things Web
            related. So, when the opportunity to jump into Full Stack
            development presented itself I was over the moon, and as a result,
            these days I'm a full-blown Stack Developer. More recently, at a
            professional level, Iâ€™ve been working with Golang on the backend
            side of things, and with React and Typescript on the front-end bits
            -{" "}
            <a
              href="https://github.com/rjmarques/personal-website"
              target="_blank"
              rel="noopener noreferrer"
            >
              this website is a SSR React App with a Typescript backend
            </a>
            .
          </p>
          <p>
            In my free time I dabble with tech I'm less familiar with. Recently
            it has been AWS ECS and Webassembly.
          </p>
        </div>
      </div>
    </div>
  </ContentSection>
);

export default Bio;
