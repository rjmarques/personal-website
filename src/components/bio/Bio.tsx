import { Col, Row } from "antd";
import React from "react";

import ContentSection from "../common/ContentSection";

import "./Bio.less";

const Bio = () => (
  <ContentSection
    title="about me"
    subTitle="Know me a bit better"
    wrapperClass="Bio"
  >
    <Row>
      <Col xs={0} sm={0} md={0} lg={4} xl={4}>
        <div className="Bio-picture">
          <img alt="bio picture" src={require("../../assets/img/me.jpg")} />{" "}
        </div>
      </Col>
      <Col xs={24} sm={24} md={24} lg={19} xl={16}>
        <div className="Bio-divider" />
        <div className="Bio-text">
          <p>
            Hi, I'm Ricardo Marques. I'm a professional nerd that works on both
            the back and front end, while using a modern tech stack to make
            things go: <i>Beep-Boop</i>.
          </p>
          <p>
            I started off, as a Java developer, building JEE applications. From
            multithread solutions, to REST integrations and even JPA enabled
            EJBs. However, I've always had a big interest in all things Web
            related. So, when the opportunity to jump into Full Stack
            development presented itself I was over the moon, and as a result,
            these days I'm a full-blown Stack Developer. More recently, at a
            professional level, Iâ€™ve been working with Typescript on the backend
            side of things, and with React on the front-end bits -{" "}
            <a
              href="https://github.com/rjmarques/personal-website"
              target="_blank"
            >
              this website was built with both those technologies
            </a>
            .
          </p>
          <p>
            On my free time I dabble with tech I don't know much about. Recently
            it has been Go and Webassembly.
          </p>
        </div>
      </Col>
    </Row>
  </ContentSection>
);

export default Bio;
