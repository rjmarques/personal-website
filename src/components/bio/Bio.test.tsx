import React from "react";
import renderer from "react-test-renderer";

import Bio from "./Bio";

describe("renders without crashing", () => {
  it("renders correctly", async () => {
    const component = renderer.create(<Bio />);

    const tree = component.toJSON();
    expect(component).toMatchSnapshot();
  });
});
