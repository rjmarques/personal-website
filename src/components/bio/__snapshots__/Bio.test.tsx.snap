// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing renders correctly 1`] = `
<section
  className="Bio"
>
  <div
    className="content"
  >
    <div
      className="Content-title"
    >
      <h2>
        about me
      </h2>
      <span
        className="Section-sub-title"
      >
        Know me a bit better
      </span>
    </div>
    <div
      className="ant-row"
    >
      <div
        className="ant-col-xs-0 ant-col-sm-0 ant-col-md-0 ant-col-lg-4 ant-col-xl-4"
      >
        <div
          className="Bio-picture"
        >
          <img
            alt="ric"
            src="me.jpg"
          />
        </div>
      </div>
      <div
        className="ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-20 ant-col-xl-20"
      >
        <div
          className="Bio-divider"
        />
        <div
          className="Bio-text materialize"
        >
          <p>
            Hi, I'm Ricardo Marques. I'm a professional nerd that works on both the back and front end, while using a modern tech stack to make things go: 
            <i>
              Beep-Boop
            </i>
            .
          </p>
          <p>
            I started off, as a Java developer, building JEE applications. From multithread solutions, to REST integrations and even JPA enabled EJBs. However, I've always had a big interest in all things Web related. So, when the opportunity to jump into Full Stack development presented itself I was over the moon, and as a result, these days I'm a full-blown Stack Developer. More recently, at a professional level, Iâ€™ve been working with Typescript on the backend side of things, and with React on the front-end bits -
             
            <a
              href="https://github.com/rjmarques/personal-website"
              rel="noopener noreferrer"
              target="_blank"
            >
              this website was built with both those technologies
            </a>
            .
          </p>
          <p>
            In my free time I dabble with tech I don't know much about. Recently it has been Go and Webassembly.
          </p>
        </div>
      </div>
    </div>
  </div>
</section>
`;
